src/item.cpp

```diff
@@ -2504,12 +2518,23 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
         }
     }
 
-    if( has_flag( "FIT" ) ) {
-        ret << _( " (fits)" );
-    }
-
-    if( has_flag( "UNDERSIZE" ) ) {
-        ret << _( " (undersized)" );
+    const bool smol = g->u.has_trait( trait_id( "SMALL2" ) ) || g->u.has_trait( trait_id( "SMALL_OK" ) );
+    const bool fits = has_flag( "FIT" );
+    const bool undersize = has_flag( "UNDERSIZE" );
+    if( fits ) {
```

Comment:
This logic is wrong, for example it prints no messages unless it's fitted.
Avoid trees, it's easier to see that it's correct like:
```
if( small && !undersize ) {
    ret << _( " (oversize)" );
} else if( !small && undersize ) {
    ret << _( " (undersize)" );
} else if( fits ) {
    ret << _( " (fits)" );
}